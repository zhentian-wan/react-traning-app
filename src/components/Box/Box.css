.Box {
    width: auto;
    min-width: 150px;
    height: auto;
    min-height: 150px;
    border: 1px solid black;
    padding: 10px;
    margin: 10px;
    cursor: pointer;
}

.red {
    background: red;
}

.green {
    background: green;
}

.highlight {
    border: 5px solid lightskyblue;
}