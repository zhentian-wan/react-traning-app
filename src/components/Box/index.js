import React from 'react';
import classnames from 'classnames';

import './Box.css';

const Box = ({color, onClick, id, highlight}) => (
  <div
      className={classnames('Box', color, {'highlight': highlight})}
      onClick={onClick}
  >
      Box2 {color} - {id}
  </div>
);

Box.propTypes = {
    onClick: React.PropTypes.func,
    color: React.PropTypes.string.isRequired,
    highlight: React.PropTypes.bool,
    id: React.PropTypes.number
};

export default Box;