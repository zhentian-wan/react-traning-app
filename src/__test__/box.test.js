
import React from 'react';
// After (15.5)
import TestUtils from 'react-dom/test-utils';
import expectLib from 'expect';
import expectJSX from 'expect-jsx';

import Box from '../components/Box';

Object.assign({}, expect, expectLib, expectJSX);

describe('box component', () => {

    test('box should be a div type', () => {
       const renderer = TestUtils.createRenderer();
       renderer.render(<Box color="red" id="0"/>);
       const result = renderer.getRenderOutput().type;
       expect(result).toEqual('div');
    });

    it('should rendering the string', () => {
       const renderer = TestUtils.createRenderer();
       renderer.render(<Box color="green" id="2" />);
       const result = renderer.getRenderOutput();
       expect(result).toMatchSnapshot()
    });

    describe('color rendering tests', () => {
        function renderingBoxThenCheckColor(color) {
            const renderer = TestUtils.createRenderer();
            renderer.render(<Box color={color} id="0"/>);
            return renderer.getRenderOutput().props.className.includes(color);
        }

        test('box should be rendered correctly: red', () => {
            expect(renderingBoxThenCheckColor('red')).toEqual(true);
        });

        test('box should be rendered correctly: green', () => {
            expect(renderingBoxThenCheckColor('green')).toEqual(true);
        });
    });


});