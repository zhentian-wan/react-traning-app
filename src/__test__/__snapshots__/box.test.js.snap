exports[`box component should rendering the string 1`] = `
<div
  className="Box green"
  onClick={undefined}>
  Box2 
  green
   - 
  2
</div>
`;
